keps/sig-api-machinery/2211-strip-managed-fields-server-side/README.md

```diff
@@ -0,0 +1,413 @@
+            }
+            a.SetManagedFields(nil)
+            return nil
+        })
+    }
+    return s.inner.Encode(obj, w)
+}
+
+func (s *serializer) Decode(data []byte, defaults *schema.GroupVersionKind, into runtime.Object) (runtime.Object, *schema.GroupVersionKind, error) {
+    return s.inner.Decode(data, defaults, into)
+}
+```
+
+3. Make `k8s.io/apimachinery/pkg/runtime/serializer.newSerializersForScheme` returns serializers for the new "unmanaged" content types.
+For instance:
+```go
+import (
+    ...
+    "k8s.io/apimachinery/pkg/runtime/serializer/unmanaged"
+)
+
+func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory, options CodecFactoryOptions) []serializerType {
+    ...
+    serializers := []serializerType{
+        jsonSerializerType,
+        {
+            AcceptContentTypes: []string{runtime.ContentTypeYAML},
+            ContentType:        runtime.ContentTypeYAML,
+            FileExtensions:     []string{"yaml"},
+            EncodesAsText:      true,
+            Serializer:         yamlSerializer,
+        },
+        {
+            AcceptContentTypes: []string{runtime.ContentTypeProtobuf},
+            ContentType:        runtime.ContentTypeProtobuf,
+            FileExtensions:     []string{"pb"},
+            Serializer:         protoSerializer,
+
+            Framer:           protobuf.LengthDelimitedFramer,
+            StreamSerializer: protoRawSerializer,
+        },
+    }
+    ...
+
+    allSerializers := append([]serializerType{}, serializers...)
+    for _, serializer := range serializers {
+        serializer.ContentType += runtime.MediaTypeSuffixOmitManagedFields
+        for i := range serializer.AcceptContentTypes {
+            serializer.AcceptContentTypes[i] += runtime.MediaTypeSuffixOmitManagedFields
+        }
+        serializer.Serializer = unmanaged.NewSerializer(serializer.Serializer)
+        serializer.StreamSerializer = unmanaged.NewSerializer(serializer.StreamSerializer)
+        serializer.PrettySerializer = unmanaged.NewSerializer(serializer.PrettySerializer)
+        allSerializers = append(allSerializers, serializer)
+    }
+    return allSerializers
+}
+```
+
+### Client-side
+
+TBD
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+TBD
```

Comment:
I know we want to merge this as provisional but for the record I want to leave the things here that I think we should make sure are tested.
- Header drops managedFields (obviously)
- Header returns 406 when not available
- Header does appropriate fallback using the `q=` mechanism.
Those tests should ensure the functionality we rely on here does not change and works as expected by apply.